### ISSUE TITLE
aurelia-binding.d.ts beta.1.0.1 issues with TypeScript noImplicitAny=true

### ISSUE THREAD

**Issue by nil4** [aurelia/binding#232]
The file [`aurelia-binding.d.ts`](https://github.com/aurelia/binding/blob/fc0af3b5f3bdf85020fce5a5ef34f85ac690cb52/dist/aurelia-binding.d.ts) at version 1.0.0-beta.1.0.1 contains the following definitions that cause errors when the `noImplicitAny` TypeScript option is enabled (note the `initialize` and `register` functions have no parameter signatures):

```typescript
  export class ValueConverterResource {
    static convention(name: string): ValueConverterResource;
    constructor(name: string);
    initialize(container, target): void;
    register(registry, name): void;
  }
 export class BindingBehaviorResource {
    static convention(name: string): BindingBehaviorResource;
    constructor(name: string);
    initialize(container, target): void;
    register(registry, name): void;
  }
```

**Comment by EisenbergEffect** [comment]
This sounds like an awesome opportunity to submit a pull request :)

**Comment by nil4** [comment]
Apologies, I am new at this GitHub thing; I wanted to create a pull request against this repository but I seem to have done something completely different. Let me try again.

**Comment by nil4** [comment]
@EisenbergEffect: pull request #234 created.

**Comment by mikeobrien** [comment]
I'm hitting another issue with 1.0.0-beta.1.0.1:

```
aurelia-binding@1.0.0-beta.1.0.1/aurelia-binding.d.ts(58,16): error TS7006: Parameter 'container' implicitly has an 'any' type.
``` 

So how are these `.d.ts` files being maintained? I was under the impression they were being generated by Babel so not something we could just fix directly in the `.d.ts` file and submit a PR for.

Also, another thought, what would you think about adding a compile step to your build to compile the `.d.ts` files? I regularly run into problems with these files when I upgrade that could have been caught with a compile step.

I'm not sure how this is being handled, so maybe missing something.

**Comment by EisenbergEffect** [comment]
The binding library doesn't use code generation for it's d.ts anymore, so you can just submit a PR against the d.ts file to fix issues.

**Comment by mikeobrien** [comment]
Thx, good to know. Is this true for all Aurelia libs or just some like binding?

**Comment by EisenbergEffect** [comment]
Only for binding at the moment. The rest have their d.ts files generated from source.

**Comment by EisenbergEffect** [comment]
We may look at converting all libraries to have custom d.ts similar to this repo in the future.

**Comment by mikeobrien** [comment]
Great, thx for the info!
